### DISCUSSION TITLE
Caching formatted result from a log call

### DISCUSSION THREAD

**Discussion by jerome-trc** [odygrd/quill#107] [Q&A] [answered]
I'm more than interested in using Quill for my project for the asynchronous formatting, but my application benefits from storing formatted log strings for display in an in-window Dear ImGui panel (not for my benefit, but that of the hypothetical user). Would there be a way to keep doing this - printing to `cout`, inserting a string to a vector, and writing to a file in one macro/call - using Quill without any hacky system stream intercepts/redirects? I haven't seen any tweak options in `Quill.h` or `TweakMe.h` or any mention of such a thing in the wiki.

**Comment by odygrd** [comment, marked as answer]
Hello, 

It is possible to implement custom handlers and register them.

Each logger can have multiple handlers attached. 
And each handler can also have its own custom format.

If you would like a single LOG_ macro call to do all the 3 things listed above you have 2 options:
1. Implement a custom `handler` that does all handing inside (file/vector/console).
2. Use the existing console and file handlers and implement a custom handler to append to a vector.

I have pushed an example that is using option 2 [here](https://github.com/odygrd/quill/blob/master/examples/example_multiple_handlers.cpp)

Keep in mind that the vector in the example above is written by a backend worker thread. If you are having a different thread reading the vector to render it to the gui then you will have to use a mutex on the vector.

Let me know if you have any issues or any other questions

**Reply by jerome-trc** [reply]
Works excellently. Thanks for the help.
